src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -912,7 +914,7 @@ public void ProduceContinue()
             }
 
             if (_httpVersion != Http.HttpVersion.Http10 &&
-                RequestHeaders.TryGetValue(HeaderNames.Expect, out var expect) &&
+                RequestHeaders!.TryGetValue(HeaderNames.Expect, out var expect) &&
```

Comment:
This should be referencing HttpRequestHeaders instead of RequestHeaders in the first place.

```suggestion
                HttpRequestHeaders.TryGetValue(HeaderNames.Expect, out var expect) &&
```